### PR TITLE
feat: support cassandra database

### PR DESCRIPTION
**Pull request by dracarys18**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,9 +39,11 @@ opentelemetry-prometheus = "0.16.0"
 prometheus = "0.13.4"
 serde = { version = "1.0.202", features = ["derive"] }
 serde_json = "1.0.117"
-masking = { git = "https://github.com/juspay/hyperswitch", tag = "v1.109.0" }
+masking = { git = "https://github.com/juspay/hyperswitch.git",branch = "update_scylla",features = ["cassandra"] }
```
Comment (SanchithHegde) [inline Cargo.toml:42]: Can we avoid the branch reference?

File: src/config.rs [hunk 2]
```diff
@@ -133,12 +136,22 @@ pub struct Config {
...
+    pub cassandra: Cassandra,
     pub log: LogConfig,
     pub pool_config: PoolConfig,
     #[cfg(feature = "mtls")]
     pub certs: Certs,
 }
 
+#[derive(Deserialize, Debug)]
+pub struct Cassandra {
+    pub known_nodes: String,
```
Comment (SanchithHegde) [inline src/config.rs:148]: Is this supposed to be a `HashSet` or a `Vec`?

File: Cargo.toml (hunk 1 above)
Comment (dracarys18) [reply Cargo.toml:42]: This is temporary will remove once the PR is merged

File: src/config.rs (hunk 2 above)
Comment (dracarys18) [reply src/config.rs]: Fixed, thanks

File: Cargo.toml (hunk 1 above)
Comment (dracarys18) [reply Cargo.toml]: Done removed

File: src/types/core/key.rs [hunk 3]
```diff
@@ -90,14 +95,45 @@ impl Display for Version {
...
+impl SerializeValue for Version {
+    fn serialize<'b>(
+        &self,
+        typ: &ColumnType<'_>,
+        writer: scylla::serialize::writers::CellWriter<'b>,
+    ) -> Result<
+        scylla::serialize::writers::WrittenCellProof<'b>,
+        scylla::serialize::SerializationError,
+    > {
+        if typ != &scylla::frame::response::result::ColumnType::Int {
```
Comment (akshay-97) [inline src/types/core/key.rs:107]: how does the type get inferred for serialize, do we need this check

**Review by akshay-97 — APPROVED** [review]

File: src/types/core/key.rs [hunk 4]
```diff
@@ -90,14 +95,45 @@ impl Display for Version {
...
+                    kind: scylla::serialize::value::BuiltinTypeCheckErrorKind::MismatchedType {
+                        expected: &[ColumnType::Int],
+                    },
+                },
+            ));
+        }
+
+        let proof = writer
+            .set_value(self.0.to_be_bytes().as_slice())
+            .map_err(|_| {
```
Comment (SanchithHegde) [inline src/types/core/key.rs:121]: We'd lose the original error information here, are we okay with this?

File: src/types/core/key.rs (hunk 3 above)
Comment (dracarys18) [reply src/types/core/key.rs:107]: Not needed this check would happen in the serde deserialization itself

**Review by jarnura — APPROVED** [review]

**Pull request merged**